From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Sat, 2 May 2020 00:31:46 +0200
Subject: luci.mk: reoder generic macros and add some comments

Define the general macros "SrcDiet", "JsMin", "CssTidy" and "SubstituteVersion"
before the special section of package luci-base.

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>

diff --git a/luci.mk b/luci.mk
index 37549f79188edbef553a30b50324ce0b8df9714a..2043ad4a62cedc4ee361b96633e9d85f0f3432cd 100644
--- a/luci.mk
+++ b/luci.mk
@@ -196,30 +196,7 @@ define Package/$(PKG_NAME)/postinst
 endef
 endif
 
-# Language selection for luci-base
-ifeq ($(PKG_NAME),luci-base)
- define Package/luci-base/config
-   config LUCI_SRCDIET
-	bool "Minify Lua sources"
-	default n
-
-   config LUCI_JSMIN
-	bool "Minify JavaScript sources"
-	default y
-
-   config LUCI_CSSTIDY
-        bool "Minify CSS files"
-        default y
-
-   menu "Translations"$(foreach lang,$(LUCI_LANGUAGES),
-
-     config LUCI_LANG_$(lang)
-	   tristate "$(shell echo '$(LUCI_LANG.$(lang))' | sed -e 's/^.* (\(.*\))$$/\1/') ($(lang))")
-
-   endmenu
- endef
-endif
-
+# some generic macros that can be used by all packages
 define SrcDiet
 	$(FIND) $(1) -type f -name '*.lua' | while read src; do \
 		if LUA_PATH="$(STAGING_DIR_HOSTPKG)/lib/lua/5.1/?.lua" luasrcdiet --noopt-binequiv -o "$$$$src.o" "$$$$src"; \
@@ -249,6 +226,30 @@ define SubstituteVersion
 	done
 endef
 
+# additional setting luci-base package
+ifeq ($(PKG_NAME),luci-base)
+ define Package/luci-base/config
+   config LUCI_SRCDIET
+	bool "Minify Lua sources"
+	default n
+
+   config LUCI_JSMIN
+	bool "Minify JavaScript sources"
+	default y
+
+   config LUCI_CSSTIDY
+        bool "Minify CSS files"
+        default y
+
+   menu "Translations"$(foreach lang,$(LUCI_LANGUAGES),
+
+     config LUCI_LANG_$(lang)
+	   tristate "$(shell echo '$(LUCI_LANG.$(lang))' | sed -e 's/^.* (\(.*\))$$/\1/') ($(lang))")
+
+   endmenu
+ endef
+endif
+
 
 LUCI_BUILD_PACKAGES := $(PKG_NAME)
 
